stdlib/src/builtin/bool.mojo

```diff
@@ -261,6 +266,15 @@ struct Bool(
             )
         )
 
+    @always_inline("nodebug")
+    fn __index__(self) -> Int:
```

Comment:
I'm not sure if we actually want this, but I could see it being useful to avoid branching in some cases?